docker/Dockerfile.cuda11

```diff
@@ -0,0 +1,32 @@
+# Use a minimal CUDA 11.8 base image
+FROM nvidia/cuda:11.8.0-base-ubuntu22.04
+
+# Set environment variables
+ENV DEBIAN_FRONTEND=noninteractive
+ENV TZ=Europe/Budapest
+
+# Install system dependencies and Python
+RUN apt-get update && apt-get install --no-install-recommends --no-install-suggests -y \
+    build-essential \
+    python3.11 \
+    python3-pip \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Install PyTorch with CUDA 11.8 support, without caching
+RUN pip3 --no-cache-dir install torch torchvision
```

Comment:
Should be:
```RUN pip3 --no-cache-dir install torch torchvision --index-url https://download.pytorch.org/whl/cu118```
Otherwise correct version is not guaranteed. Alternatively, we could just use version specifiers e.g. ```torch==2.4.1+cu118```